src/components/EmojiPicker/EmojiPickerMenu/EmojiPickerMenu.jsx

```diff
@@ -0,0 +1,554 @@
+            const isHeader = (e) => e.header || e.spacer;
+            do {
+                newIndex += steps;
+                if (newIndex < 0) {
+                    break;
+                }
+            } while (isHeader(filteredEmojis[newIndex]));
+        };
+
+        switch (arrowKey) {
+            case 'ArrowDown':
+                move(CONST.EMOJI_NUM_PER_ROW, () => highlightedIndex + CONST.EMOJI_NUM_PER_ROW > filteredEmojis.length - 1);
+                break;
+            case 'ArrowLeft':
+                move(
+                    -1,
+                    () => highlightedIndex - 1 < firstNonHeaderIndex.current,
+                    () => {
+                        // Reaching start of the list, arrow left set the focus to searchInput.
+                        focusInputWithTextSelect();
+                        newIndex = -1;
+                    },
+                );
+                break;
+            case 'ArrowRight':
+                move(1, () => highlightedIndex + 1 > filteredEmojis.length - 1);
+                break;
+            case 'ArrowUp':
+                move(
+                    -CONST.EMOJI_NUM_PER_ROW,
+                    () => highlightedIndex - CONST.EMOJI_NUM_PER_ROW < firstNonHeaderIndex.current,
+                    () => {
+                        // Reaching start of the list, arrow up set the focus to searchInput.
+                        focusInputWithTextSelect();
+                        newIndex = -1;
+                    },
+                );
+                break;
+            default:
+                break;
+        }
+
+        // Actually highlight the new emoji, apply keyboard movement styles, and disable pointer events
+        if (newIndex !== highlightedIndex) {
+            setHighlightedIndex(newIndex);
+            setArePointerEventsDisabled(true);
+            setIsUsingKeyboardMovement(true);
+        }
+    }
+
+    function scrollToHeader(headerIndex) {
+        const calculatedOffset = Math.floor(headerIndex / CONST.EMOJI_NUM_PER_ROW) * CONST.EMOJI_PICKER_HEADER_HEIGHT;
+        emojiListRef.current?.flashScrollIndicators();
+        emojiListRef.current?.scrollToOffset({offset: calculatedOffset, animated: true});
+    }
+
+    /**
+     * Filter the entire list of emojis to only emojis that have the search term in their keywords
+     *
+     * @param {String} searchTerm
+     */
+    const filterEmojis = useCallback(
+        _.debounce((searchTerm) => {
+            const normalizedSearchTerm = searchTerm.toLowerCase().trim().replaceAll(':', '');
+            emojiListRef.current?.scrollToOffset({offset: 0, animated: false});
+            if (normalizedSearchTerm === '') {
+                // There are no headers when searching, so we need to re-make them sticky when there is no search term
+                setFilteredEmojis(emojis.current);
+                setHeaderIndices(headerRowIndices.current);
```

Comment:
This line will be deleted
(`setHeaderIndices(headerRowIndices.current);`)

>This could be deleted because in the entire file, `headerRowIndices.current` is NOT assigned a value _separately from the state_. 
>i.e If  `headerRowIndices.current` ref is assigned a value, `headerIndices` is also assigned that same new value (always)